crux_core/src/steps.rs

```diff
@@ -1,143 +1,107 @@
-use std::{
-    collections::{hash_map::Entry, HashMap},
-    fmt,
-    sync::Mutex,
-};
+use std::fmt::{self, Debug};
+use thiserror::Error;
 
-use uuid::Uuid;
+use crate::capability::Operation;
 
-use crate::Request;
-
-pub(crate) struct Step<T> {
-    pub(crate) payload: T, // T is an Operation first, then Effect once mapped
-    pub(crate) resolve: Option<Resolve>,
-}
-
-type OnceFn = dyn FnOnce(&[u8]) + Send;
-type ManyFn = dyn Fn(&[u8]) -> Result<(), ()> + Send;
-
-pub(crate) enum Resolve {
-    Once(Box<OnceFn>),
-    Many(Box<ManyFn>),
-}
+pub struct Step<Op>(pub Op, pub Resolve<Op::Output>)
```

Comment:
Not sure about `pub Op` & `pub Resolve`.  It does destructure nice, but means adding another field is a breaking change, and when you're not destructuring `step.0` & `step.1` are not very clear.

Wondering if we'd be better making them private and exposing via functions?  Or making it a struct w/ public fields?  Not sure.

Reply:
I'll have another look at the Rust API surface in another PR, once the mechanics all work. 

I think there's a few renames and ergonomic improvements to be made before calling it a public API.